AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -899,6 +890,51 @@ - (void)transitionContext:(_ASTransitionContext *)context didComplete:(BOOL)didC
   _transitionContext = nil;
 }
 
+#pragma mark - Layout
+
+- (void)_completePendingLayoutTransition
+{
+  ASDN::MutexLocker l(_propertyLock);
+  if (_pendingLayoutTransition) {
+    [self _setCalculatedLayout:_pendingLayoutTransition.pendingLayout];
+    [self _completeTransition:_pendingLayoutTransition];
+    _pendingLayoutTransition = nil;
+  }
+  [self _completeLayoutCalculation];
+}
+
+- (void)_setCalculatedLayout:(ASLayout *)layout
+{
+  ASDN::MutexLocker l(_propertyLock);
+  
+  ASDisplayNodeAssertTrue(layout.layoutableObject == self);
+  ASDisplayNodeAssertTrue(layout.size.width >= 0.0);
+  ASDisplayNodeAssertTrue(layout.size.height >= 0.0);
+  
+  _layout = layout;
+}
+
+- (void)_completeTransition:(ASLayoutTransition *)layoutTransition
+{
+  // Layout transition is not supported for non implicit hierarchy managed nodes
+  if (layoutTransition == nil || self.usesImplicitHierarchyManagement == NO) {
+    return;
+  }
+
+  // Trampoline to the main thread if necessary
+  if (ASDisplayNodeThreadIsMain() == NO && layoutTransition.isSynchronous == NO) {
+
+    // Subnode insertions and removals need to happen always on the main thread if at least one subnode is already loaded
+    ASPerformBlockOnMainThread(^{
+      [layoutTransition completeTransition];
+    });
+    
+    return;
```

Comment:
This early return format is very confusing IMO.  The next line should be in an else clause, with no early return in this clause.

I'd suggest flipping the condition so that the case without the dispatch to main is first, in the if, and then the block dispatch one is in the else.